akka-http/src/main/scala/akka/http/javadsl/server/directives/PathDirectives.scala

```diff
@@ -136,52 +66,213 @@ abstract class PathDirectives extends MiscDirectives {
+    D.pathPrefixTest(pm.toScala) { case (t1, t2) ⇒ inner.apply(t1, t2).toScala }
+  }
+
+  /**
+   * Checks whether the unmatchedPath of the [[RequestContext]] has a prefix matched by the
+   * given PathMatcher. However, as opposed to the `pathPrefix` directive the matched path is not
+   * actually "consumed".
+   */
+  def rawPathPrefixTest(segment: String, inner: Supplier[Route]): Route = ScalaRoute {
+    D.rawPathPrefixTest(segment) { inner.get().toScala }
+  }
+  def rawPathPrefixTest(pm: PathMatcher0, inner: Supplier[Route]): Route = ScalaRoute {
+    D.rawPathPrefixTest(pm.toScala) { inner.get().toScala }
+  }
+  def rawPathPrefixTest[T1](pm: PathMatcher1[T1], inner: Function[T1, Route]): Route = ScalaRoute {
+    D.rawPathPrefixTest(pm.toScala) { t1 ⇒ inner.apply(t1).toScala }
+  }
+  def rawPathPrefixTest[T1, T2](pm: PathMatcher2[T1, T2], inner: BiFunction[T1, T2, Route]): Route = ScalaRoute {
+    D.rawPathPrefixTest(pm.toScala) { case (t1, t2) ⇒ inner.apply(t1, t2).toScala }
+  }
+
+  /**
+   * Applies the given [[PathMatcher]] to a suffix of the remaining unmatchedPath of the [[RequestContext]].
+   * If matched the value extracted by the [[PathMatcher]] is extracted and the matched parts of the path are consumed.
+   * Note that, for efficiency reasons, the given [[PathMatcher]] must match the desired suffix in reversed-segment
+   * order, i.e. `pathSuffix("baz" / "bar")` would match `/foo/bar/baz`!
+   */
+  def pathSuffix(segment: String, inner: Supplier[Route]): Route = ScalaRoute {
+    D.pathSuffix(segment) { inner.get().toScala }
+  }
+  def pathSuffix(pm: PathMatcher0, inner: Supplier[Route]): Route = ScalaRoute {
+    D.pathSuffix(pm.toScala) { inner.get().toScala }
+  }
+  def pathSuffix[T1](pm: PathMatcher1[T1], inner: Function[T1, Route]): Route = ScalaRoute {
+    D.pathSuffix(pm.toScala) { t1 ⇒ inner.apply(t1).toScala }
+  }
+  def pathSuffix[T1, T2](pm: PathMatcher2[T1, T2], inner: BiFunction[T1, T2, Route]): Route = ScalaRoute {
+    D.pathSuffix(pm.toScala) { case (t1, t2) ⇒ inner.apply(t1, t2).toScala }
+  }
+
+  /**
+   * Checks whether the unmatchedPath of the [[RequestContext]] has a suffix matched by the
+   * given PathMatcher. However, as opposed to the pathSuffix directive the matched path is not
+   * actually "consumed".
+   * Note that, for efficiency reasons, the given PathMatcher must match the desired suffix in reversed-segment
+   * order, i.e. `pathSuffixTest("baz" / "bar")` would match `/foo/bar/baz`!
+   */
+  def pathSuffixTest(segment: String, inner: Supplier[Route]): Route = ScalaRoute {
+    D.pathSuffixTest(segment) { inner.get().toScala }
+  }
+  def pathSuffixTest(pm: PathMatcher0, inner: Supplier[Route]): Route = ScalaRoute {
+    D.pathSuffixTest(pm.toScala) { inner.get().toScala }
+  }
+  def pathSuffixTest[T1](pm: PathMatcher1[T1], inner: Function[T1, Route]): Route = ScalaRoute {
+    D.pathSuffixTest(pm.toScala) { t1 ⇒ inner.apply(t1).toScala }
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Non ascii characters are not allowed](https://www.codacy.com/app/vasseurjb/akka/file/2046270054/issues/source?bid=3132306&fileBranchId=3132379#l210)